src/main/java/com/powsybl/openloadflow/sa/AcSecurityAnalysis.java

```diff
@@ -174,23 +174,46 @@ private static void findAllSwitchesToOperate(Network network, List<Action> actio
                 });
     }
 
-    private static Map<String, LfAction> createLfActions(LfNetwork network, List<Action> actions) {
+    private static Map<String, LfAction> createLfActions(LfNetwork network, Set<Action> actions) {
         return actions.stream()
                 .map(action -> LfAction.create(action, network))
                 .flatMap(Optional::stream)
                 .collect(Collectors.toMap(LfAction::getId, Function.identity()));
     }
 
+    private static Map<String, Action> indexActionsById(List<Action> actions) {
+        return actions.stream()
+                .collect(Collectors.toMap(
+                    Action::getId,
+                    Function.identity(),
+                    (action1, action2) -> {
+                        throw new PowsyblException("An action '" + action1.getId() + "' already exist");
+                    }
+                ));
+    }
+
     private static Map<String, List<OperatorStrategy>> indexOperatorStrategiesByContingencyId(List<PropagatedContingency> propagatedContingencies,
-                                                                                              List<OperatorStrategy> operatorStrategies) {
+                                                                                              List<OperatorStrategy> operatorStrategies,
+                                                                                              Map<String, Action> actionsById,
+                                                                                              Set<Action> neededActions) {
```

Comment:
What I understand from previous talk with @sylvlecl and you is that the actions order matters. That is why I used a list instead of a set. With your modifications, are we sure that order is conserved?